File: Data Structures & Algorithms/buy-and-sell-crypto/submission-18.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        l, r = 0, 0

        while r < len(prices):
            if prices[l] <= prices[r]:  
                maxP = max(maxP, prices[r] - prices[l])   
                r += 1           
            else:
                l = r
        return maxP

File: Data Structures & Algorithms/trapping-rain-water/submission-18.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        let res = 0
        let l = 0, r = height.length - 1
        let maxleft = height[l], maxright = height[r]
        while (l < r) {
            if (height[l] <= height[r]) {
                l++
                maxleft = Math.max(maxleft, height[l])
                res += maxleft - height[l]
            } else {
                r--
                maxright = Math.max(maxright, height[r])
                res += maxright - height[r]
            }
        }
        return res
    }
}
